### PR TITLE
Appendix A: add source code for all examples

### PR DESCRIPTION
**Pull request by Peter-VanWeert**

### CODE REVIEW

File: Examples/NoModules/Appendix A/ExA_01/ExA_01.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Defining object-like macros
+import <iostream>;
```
Comment (MarcGregoire) [inline Examples/NoModules/Appendix A/ExA_01/ExA_01.cpp:2]: Should be:
```
#include <iostream>
```

File: Examples/NoModules/Appendix A/ExA_03/ExA_03.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Using std::source_location, C++'s answer to __LINE__ and __FILE__
```
Comment (MarcGregoire) [inline Examples/NoModules/Appendix A/ExA_03/ExA_03.cpp:1]: Couldn't compile with VC++, doesn't support source_location yet.

Comment (MarcGregoire) [reply Examples/NoModules/Appendix A/ExA_03/ExA_03.cpp:1]: but code looks ok.

File: Examples/NoModules/Appendix A/ExA_04/ExA_04.cpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Demonstrating assertions
+#include <iostream>
+#include <format>
+#include <cassert> 
+
+// Uncomment to disable the assertion in main()
+//#define NDEBUG
```
Comment (MarcGregoire) [inline Examples/NoModules/Appendix A/ExA_04/ExA_04.cpp:7]: Even when uncommenting, the assert is still triggered.
You need to move this line to before you include cassert.

File: Examples/NoModules/Appendix A/ExA_05/ExA_05.cpp [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// ExA_05.cpp
+// Introducing the #include directive
+import <iostream>;
```
Comment (MarcGregoire) [inline Examples/NoModules/Appendix A/ExA_05/ExA_05.cpp:3]: #include <iostream>

File: Examples/NoModules/Appendix A/ExA_06/ExA_06.cpp [hunk 5]
```diff
@@ -0,0 +1,30 @@
+// Defining the same function twice
```
Comment (MarcGregoire) [inline Examples/NoModules/Appendix A/ExA_06/ExA_06.cpp:1]: Should you put a warning here that this file as-is will not compile?

File: Examples/NoModules/Appendix A/ExA_10/BadMath.h [hunk 6]
```diff
@@ -0,0 +1,7 @@
+auto square(const auto& x) { return x * x; }   // An abbreviated function template
+
+const double lambda = 1.303577269034296391257; // Conway's constant
```
Comment (MarcGregoire) [inline Examples/NoModules/Appendix A/ExA_10/BadMath.h:3]: uniform init

File: Examples/NoModules/Appendix A/ExA_10A/BetterMath.h [hunk 7]
```diff
@@ -0,0 +1,13 @@
+// A second, better attempt at creating a header file
+#ifndef BETTER_MATH_H
+#define BETTER_MATH_H
+
+auto square(const auto& x) { return x * x; }   // An abbreviated function template
+
+const double lambda = 1.303577269034296391257; // Conway's constant
```
Comment (MarcGregoire) [inline Examples/NoModules/Appendix A/ExA_10A/BetterMath.h:7]: uniform init

File: Examples/NoModules/Appendix A/ExA_11/BetterMath.h [hunk 8]
```diff
@@ -0,0 +1,13 @@
+// A second, better attempt at creating a header file
+#ifndef BETTER_MATH_H
+#define BETTER_MATH_H
+
+auto square(const auto& x) { return x * x; }   // An abbreviated function template
+
+const double lambda = 1.303577269034296391257; // Conway's constant
```
Comment (MarcGregoire) [inline Examples/NoModules/Appendix A/ExA_11/BetterMath.h:7]: uniform init

File: Examples/NoModules/Appendix A/ExA_12/ProperMath.h [hunk 9]
```diff
@@ -0,0 +1,13 @@
+// Your first, proper header file
+#ifndef PROPER_MATH_H
+#define PROPER_MATH_H
+
+auto square(const auto& x) { return x * x; }   // An abbreviated function template
+
+const inline double lambda = 1.303577269034296391257; // Conway's constant
```
Comment (MarcGregoire) [inline Examples/NoModules/Appendix A/ExA_12/ProperMath.h:7]: uniform init

File: Examples/NoModules/Appendix A/ExA_13/Box.cpp [hunk 10]
```diff
@@ -0,0 +1,15 @@
+#include "Box.h"
+
+Box::Box() : Box(1.0, 1.0, 1.0) {}
```
Comment (MarcGregoire) [inline Examples/NoModules/Appendix A/ExA_13/Box.cpp:3]: uniform init

File: Examples/NoModules/Appendix A/ExA_13/Box.cpp [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+Box::Box(double length, double width, double height)
+  : m_length(length), m_width(width), m_height(height)
```
Comment (MarcGregoire) [inline Examples/NoModules/Appendix A/ExA_13/Box.cpp:5]: uniform init

File: Examples/NoModules/Appendix A/ExA_13A/Box.h [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+private:
+  double m_length {1.0};
+  double m_width {1.0};
+  double m_height {1.0};
+};
+
+// Out-of-class member definitions must be explicitly marked as inline 
+inline Box::Box(double length, double width, double height)
+  : m_length(length), m_width(width), m_height(height)
```
Comment (MarcGregoire) [inline Examples/NoModules/Appendix A/ExA_13A/Box.h:28]: uniform init

**Review by MarcGregoire — CHANGES_REQUESTED** [review]

File: Examples/NoModules/Appendix A/ExA_06/ExA_06.cpp (hunk 5 above)
Comment (Peter-VanWeert) [reply Examples/NoModules/Appendix A/ExA_06/ExA_06.cpp:1]: The reader should now if he's following the book... But sure, why not.

**Pull request merged**